variant/JsonStrings.js

```diff
@@ -0,0 +1,61 @@
+// The parseParameter function will typically be called for every page and 
+// the setParameter function is called by each active plugin to bundle specific attacks
+
+// Note that new custom input vector scripts will initially be disabled
+// Right click the script in the Scripts tree and select "enable"  
+
+// The following handles differences in printing between Java 7's Rhino JS engine
+// and Java 8's Nashorn JS engine
+if (typeof println == 'undefined') this.println = print;
+
+/*
+This variant script can be used as an alternative to the default JSON input vectors
+when one only wants to scan the string fields of the JSON object, not integers.
+
+Input vectors will be located and set recursively and objects containing up to three
+layers of objects, lists and strings seem to work as intended.
+*/
+
+function parseParameters(helper, msg) {
+	//we're only interested in JSON requests
+	if (msg.getRequestHeader().getHeader("Content-Type") != 'application/json'){
+		return;
+	}
+
+	body = msg.getRequestBody().toString();
+	if (body == "") {return;}
+	obj = JSON.parse(body)
+	recursive_parse(helper, obj, []);
+}
+
+function recursive_parse(helper, obj, path){
+	if (typeof(obj) == "number"){
+		//skipping this is why I wrote this script!
+	}else if(typeof(obj) == "string"){
+		//print(JSON.stringify(path) + " : " + obj)
+		//here the parameter is added with the name being a JSON
+		//list of keys that act as the JSON path to the input vector
+		helper.addParamPost(JSON.stringify(path), obj);
```

Comment:
That's another case, I was not referring to the UI/Alert data but the behaviour of the input vector script, both name and value can be changed by the scanners, so, if a scanner sets the name to other thing it will break this input vector script, but again there's no workaround to properly identify the parameter being changed.

Yes, printing something (or log) and return is the best that the script can do IMO, the alternative is to just let ZAP disable the script because of the error.